general/src/types/impls.rs

```diff
@@ -420,7 +415,11 @@ fn datetime_to_timestamp(dt: DateTime<Utc>) -> f64 {
     // If we want to support nanoseconds at some point we will probably have to start using strings
     // everywhere. Even then it's unclear how to deal with it in Python code as a `datetime` cannot
     // store nanoseconds.
-    (timestamp * 1_000_000f64).round() / 1_000_000f64
+    //
+    // We use `timestamp_subsec_nanos` instead of `timestamp_subsec_micros` anyway to get better
+    // rounding behavior.
+    let micros = (f64::from(dt.timestamp_subsec_nanos()) / 1_000f64).round();
+    dt.timestamp() as f64 + (micros / 1_000_000f64)
```

Comment:
I thought of
```rust
(dt.timestamp() * 1_000_000f64 + f64::from(dt.timestamp_subsec_nanos()) / 1_000f64).round() / 1_000_000f64
```

but i wasn't sure if my suggestion was better.

Something like https://github.com/programmerjake/algebraics would deal with the float point error stuff 😄